yield_plugin/src/yield_plugin.cpp

```diff
@@ -659,22 +662,35 @@ namespace yield_plugin
         double vehicle_downtrack = wm_->routeTrackPos(vehicle_point).downtrack;
         double object_downtrack = wm_->routeTrackPos(object_point).downtrack;
 
-        if (vehicle_downtrack > object_downtrack + config_.vehicle_length / 2)  // if half a length of the vehicle past, it is considered behind
+        if (vehicle_downtrack > object_downtrack + config_.vehicle_length / 2) // if half a length of the vehicle past, it maybe considered behind
```

Comment:
> I'm not sure I understand the consecutive_clearance_count solution. It appears we are keeping track of how often a specific obstacle is beyond the collision threshold and assuming this count somehow translates to an obstacle being behind the vehicle. Is this correct?

It is actually checking a case where the obstacle is within collision threshold. It would be detected a collision all the time without this check even in cases where vehicle is passed the pedestrian

> Perhaps we could do an alternative implementation based on a line passing through the vehicle's rear axis. If the obstacle's position is on one side of the line (I believe this would be the epigraph) then it is not behind the vehicle. Otherwise, it is behind the vehicle.

Current implementation is exactly doing that. vehicle_downtrack and object_downtrack are distances from the vehicle's engagement position projected from their x,y position to along the road. So in this case, the line passing through vehicle's rear axis would be vehicle_downtrack. And if object downtrack is more than half of the vehicle_length (to be conservative) away from the vehicle_downtrack, it means it is passed.